### PR TITLE
CAD-878: Trace node up time.

### PR DESCRIPTION
**Pull request by denisshevchenko**
Issue
-----------

- Trace node's real up time.

- This PR **does not result** in breaking changes to upstream dependencies.

Checklist
---------
- [ ] This PR contains all the work required to resolve the linked issue.

- [ ] The work contained has sufficient documentation to describe what it does and how to do it.

- [ ] The work has sufficient tests and/or testing.

- [ ] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [x] I have added the appropriate labels to this PR.

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Run.hs [hunk 1]
```diff
@@ -33,6 +33,7 @@ import qualified Data.List as List
 import           Data.Proxy (Proxy (..))
 import           Data.Semigroup ((<>))
 import           Data.Text (Text, breakOn, pack, take)
+import           Data.Time.Clock (UTCTime, diffUTCTime, getCurrentTime)
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:36]:
```suggestion
import           GHC.Clock (getMonotonicTimeNSec)
```

File: cardano-node/src/Cardano/Node/Run.hs [hunk 2]
```diff
@@ -117,7 +119,13 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
     tracers <- mkTracers (ncTraceConfig nc) trace
 
     case ncViewMode nc of
-      SimpleView -> handleSimpleNode p trace tracers npm (const $ pure ())
+      SimpleView -> do
+        nodeLaunchTime <- getCurrentTime
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:123]: use `getMonotonicTimeNSec`

File: cardano-node/src/Cardano/Node/Run.hs [hunk 3]
```diff
@@ -145,6 +156,21 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
       hn0 <- pack <$> getHostName
       return $ take 8 $ fst $ breakOn "." hn0
 
+-- | The node sends its real up time, every second.
+traceNodeUpTime
+  :: Trace IO Text
+  -> UTCTime
+  -> IO ()
+traceNodeUpTime tr nodeLaunchTime = do
+  now <- getCurrentTime
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:165]:
```suggestion
  now <- getMonotonicTimeNSec
```

File: cardano-node/src/Cardano/Node/Run.hs [hunk 4]
```diff
@@ -145,6 +156,21 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
...
+  let upTime = toRational $ diffUTCTime now nodeLaunchTime
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:166]:
```suggestion
  let upTimeNanosecs = now - nodeLaunchTime
```

File: cardano-node/src/Cardano/Node/Run.hs [hunk 5]
```diff
@@ -145,6 +156,21 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
       hn0 <- pack <$> getHostName
       return $ take 8 $ fst $ breakOn "." hn0
 
+-- | The node sends its real up time, every second.
+traceNodeUpTime
+  :: Trace IO Text
+  -> UTCTime
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:162]:
```suggestion
  -> Word64
```

File: cardano-node/src/Cardano/Node/Run.hs [hunk 6]
```diff
@@ -145,6 +156,21 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
...
+  -> IO ()
+traceNodeUpTime tr nodeLaunchTime = do
+  now <- getCurrentTime
+  let upTime = toRational $ diffUTCTime now nodeLaunchTime
+      upTimeInSec = (fromIntegral $ numerator upTime) / (fromIntegral $ denominator upTime)
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:167]:
```suggestion
```

File: cardano-node/src/Cardano/Node/Run.hs [hunk 7]
```diff
@@ -145,6 +156,21 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
...
+  meta <- mkLOMeta Notice Public
+  let tr' = appendName "upTime" tr
+  traceNamedObject tr' (meta, LogValue "upTime" (PureD upTimeInSec))
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:170]:
```suggestion
  traceNamedObject tr' (meta, LogValue "upTime" (Nanoseconds upTimeNanosecs))
```
so we keep its type

**Comment by denisshevchenko** [comment]
@CodiePP Fixed.

**Pull request merged**